### ISSUE TITLE
Implement UGRID default start_index

### ISSUE THREAD

**Issue by pp-mo** [SciTools/iris#4214]
## ✨ Feature Request
In the UGRID conventions, the 'start_index' property is never actually required,
e.g. [in this section](https://ugrid-conventions.github.io/ugrid-conventions/#2d-triangular-mesh-topology)
> For the indexing one may use either 0- or 1-based indexing; the convention used should be specified using a start_index attribute to the index variable (i.e. Mesh2_face_nodes in the example below). Consistent with the CF-conventions compression option, the **connectivity indices are 0-based by default**. See [this section on zero or one-based indexing](https://ugrid-conventions.github.io/ugrid-conventions/#zero-or-one) for more details.

But in our current code, we have [assumed it is always there](https://github.com/SciTools/iris/blob/mesh-data-model/lib/iris/experimental/ugrid/__init__.py#L3646)

Notable specifically for connectivities, 
might possibly apply elsewhere, but at present I think not : probably just in this one place ?

**Comment by trexfeathers** [comment]
This should _I think_ be fine to implement - the `Connectivity` class always has a `start_index` attribute, but defaults to `=0` in line with the UGRID conventions.

**Comment by pp-mo** [comment]
>  fine to implement

Absolutely, logically it is all fine, we must just stop assuming that the file variables always has a start_index.

So, I think [this PR](https://github.com/SciTools/iris/pull/4224) should be all that's needed

**Comment by trexfeathers** [comment]
Is there a specific reason why this hasn't happened yet? Has it just been forgotten?
